erpnext/accounts/doctype/process_deferred_accounting/process_deferred_accounting.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _
+from frappe.model.document import Document
+from erpnext.accounts.deferred_revenue import convert_deferred_expense_to_expense, \
+	convert_deferred_revenue_to_income
+
+class ProcessDeferredAccounting(Document):
+	def autoname(self):
+		naming_series = [self.type, self.company, self.account, self.posting_date]
+		self.name = '-'.join(filter(None, naming_series))
+
+	def on_submit(self):
+		conditions = self.build_conditions()
+		if self.type == 'Income':
+			convert_deferred_revenue_to_income(self.start_date, self.end_date, conditions)
+		else:
+			convert_deferred_expense_to_expense(self.start_date, self.end_date, conditions)
+
+	def build_conditions(self):
+		conditions=''
+		if self.account:
+			conditions += "AND item.deferred_revenue_account='%s'"%(self.account)
```

Comment:
What is account is selected for type "Expense". Also consider `deferred_expense_account` here. Also add a on_cancel function to handle cancellation of this doc.

Reply:
fixed the deferred revenue type